crypto/sign.go

```diff
@@ -42,9 +42,9 @@ func VerifySignature(pubKey, message, signature []byte) error {
 
 // Sign the document with the private key and return the signature along with the public key for the verification
 // assumes that signing root for the document is generated
-func Sign(centIDBytes []byte, privateKey []byte, pubKey []byte, payload []byte) *coredocumentpb.Signature {
+func Sign(id []byte, privateKey []byte, pubKey []byte, payload []byte) *coredocumentpb.Signature {
```

Comment:
Keytools shouldn't deal with creating protobuf messages IMO. I would prefer to move this out of the keytools package. I think this would be a good issue to add to house-keeping and address later.

Reply:
agree

Reply:
We are going to use the sign method that is part of the Account (all keys live in that context), there is a function already in the package. It is debatable to return the Signature proto there or just a byte array but I like the idea of having the signing as part of the account struct.